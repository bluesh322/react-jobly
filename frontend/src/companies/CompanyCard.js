import React from "react";
import { Link } from "react-router-dom";
import {
  Card,
  CardContent,
  CardActions,
  Button,
  Typography,
  makeStyles,
  Box,
} from "@material-ui/core";

const useStyles = makeStyles({
  root: {
    minWidth: 200,
  },

  title: {
    fontSize: 16,
  },
});

const CompanyCard = ({ handle, name, description, numEmployees, logoUrl }) => {
  const classes = useStyles();

  return (
    <Box my={2}>
      <Link to={`companies/${handle}`}>
        <Card className={classes.root}>
          <CardContent>
            <Typography component="h2" className={classes.title}>
              {name}
            </Typography>
            <Typography component="h4">{description}</Typography>
          </CardContent>
          <CardActions>
            <Button>{logoUrl && <img src={logoUrl} alt={name} />}</Button>
          </CardActions>
        </Card>
      </Link>
    </Box>
  );
};

export default CompanyCard;
