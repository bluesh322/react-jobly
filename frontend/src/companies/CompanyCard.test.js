import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

test("matches snapshot with logo", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard
        handle="rithm"
        name="Rithm School"
        description="Become an exceptional developer in 16 weeks."
        logo_url="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
      ></CompanyCard>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

test("matches snapshot without logo", async () => {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard
        handle="algo"
        name="Algo School"
        description="Become a mediocre developer in 160 weeks."
      ></CompanyCard>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
