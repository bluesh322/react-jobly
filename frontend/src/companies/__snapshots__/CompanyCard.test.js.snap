// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <a
      href="/companies/rithm"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root makeStyles-title-2 MuiTypography-body1"
          >
            Rithm School
          </h2>
          <h4
            class="MuiTypography-root MuiTypography-body1"
          >
            Become an exceptional developer in 16 weeks.
          </h4>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-6"
  >
    <a
      href="/companies/algo"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-4 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root makeStyles-title-5 MuiTypography-body1"
          >
            Algo School
          </h2>
          <h4
            class="MuiTypography-root MuiTypography-body1"
          >
            Become a mediocre developer in 160 weeks.
          </h4>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
